zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -1126,34 +1164,49 @@ pub(crate) fn transaction_data_requests<P: consensus::Parameters>(
     let chain_tip_height =
         super::chain_tip_height(conn)?.ok_or(SqliteClientError::ChainHeightUnknown)?;
 
+    debug!(
+        "Generating transaction data requests as of chain tip height {}",
+        chain_tip_height
+    );
+
     // We cannot construct address-based transaction data requests for the case where we cannot
     // determine the height at which to begin, so we require that either the target height or mined
     // height be set.
     let mut spend_requests_stmt = conn.prepare_cached(
         "SELECT
             ssq.address,
-            IFNULL(t.target_height, t.mined_height)
+            COALESCE(tro.max_observed_unspent_height, t.mined_height, t.target_height)
          FROM transparent_spend_search_queue ssq
          JOIN transactions t ON t.id_tx = ssq.transaction_id
-         WHERE t.target_height IS NOT NULL
-         OR t.mined_height IS NOT NULL",
+         JOIN transparent_received_outputs tro ON tro.transaction_id = t.id_tx
+         LEFT OUTER JOIN transparent_received_output_spends tros
+            ON tros.transparent_received_output_id = tro.id
+         WHERE tros.transaction_id IS NULL
+         AND (t.target_height IS NOT NULL OR t.mined_height IS NOT NULL)
+         AND (
+             tro.max_observed_unspent_height IS NULL
+             OR tro.max_observed_unspent_height < :chain_tip_height
```

Comment:
I think this needs to be `<=` given the (current) definition of `tro.max_observed_unspent_height` as being unspent at the start of the specified block (corresponding to setting it to `checked_at + 1` in my other comment)? Otherwise we don't request a TDR unless we're at least 2 blocks outdated.